src/registrar/models/contact.py

```diff
@@ -54,6 +54,50 @@ class Contact(TimeStampedModel):
         db_index=True,
     )
 
+    def has_more_than_one_join(self, expected_relation):
+        """Helper for finding whether an object is joined more than once.
+        expected_relation is the one relation with one expected join"""
+        # all_relations is the list of all_relations (from contact) to be checked for existing joins
+        all_relations = [
+            "user",
+            "authorizing_official",
+            "submitted_applications",
+            "contact_applications",
+            "information_authorizing_official",
+            "submitted_applications_information",
+            "contact_applications_information",
+        ]
+        return any(self._has_more_than_one_join_per_relation(rel, expected_relation) for rel in all_relations)
```

Comment:
(Suggestion to try) I think there is a "is_relation" flag on the [model field reference](https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.is_relation). If you iterate through _meta you may be able to dynamically infer these fields. I would recommend that if you can as that would be more resilient to model changes, but not blocking if this doesn't work

def get_all_relations(self):
    return [f.name for f in self._meta.get_fields() if f.is_relation]

Reply:
This is great.  Thank you.  Incorporated this.